### ISSUE TITLE
Responses to messages are sequential since 0.19

### ISSUE THREAD

**Issue by enver** [DandyDev/slack-machine#414]
With recent changes slack-machine responses became sequential. Back in version 0.18.2 responses were processed in parallel and did not block each other. Currently, in case response method takes a significant amount of time to complete all other requests are queued and executed after a long-running response has completed.

Reproduction steps:
Define a simple sleep method:
```
    ###############################################################################
    @respond_to(r'^\s*sleep\s+(?P<sleep_time>\d+)\s*$', re.IGNORECASE)
    def sleep(self, message, sleep_time):
        message.reply('Sleeping: %s' % sleep_time)
        time.sleep(int(sleep_time))
        message.reply('Woken up')
```

And invoke it twice:
-> sleep 20
<- Sleeping: 20
-> sleep 20
<- Woken up
<- Sleeping: 20
<- Woken up

Expected result:
-> sleep 20
<- Sleeping: 20
-> sleep 20
<- Sleeping: 20
<- Woken up
<- Woken up

It seems that changes [here](https://github.com/DandyDev/slack-machine/blob/97f9693652d4d6453661f87fea01b62c041bd810/machine/dispatch.py#L31): cause this behaviour

**Comment by DandyDev** [comment]
This is because of fundamental changes in the underlying slacklclient library v2, especially in the RTM part. The slackclient library is now taking care of dispatching functions based on incoming RTM events, so there is no easy way to parallelize is like I did before.
The slackclient library _does_ support asyncio, and I want to support that as well, but it would require significant rewrites of slack-machine.

I also noticed that there has even been _another_ big update in the sense that slackclient has been deprecated in favour of [python-slack-sdk](https://github.com/slackapi/python-slack-sdk). I have to look into that and start an upgrade path for slack-machine, but I won't have a lot of time in the coming weeks/months to pull this off.

PRs are welcome!
